ArkKit/ark-physics-kit/ArkPhysicsSystem.swift

```diff
@@ -0,0 +1,232 @@
+import Foundation
+
+/**
+ *  A system which updates all running animation instances in ArkECS after the given delta.
+ */
+class ArkPhysicsSystem: System {
+    var active: Bool
+    var scene: AbstractArkGameScene
+    var eventManager: ArkEventManager
+
+    init(active: Bool = true, gameScene: AbstractArkGameScene, eventManager: ArkEventManager) {
+        self.active = active
+        self.scene = gameScene
+        self.eventManager = eventManager
+    }
+    
+    func update(deltaTime: TimeInterval, arkECS: ArkECS) {
+        let physicsComponents = getPhysicsComponents(arkECS)
+        syncToPhysicsEngine(physicsComponents, arkECS: arkECS)
+        scene.update(deltaTime)
+        syncFromPhysicsEngine(arkECS: arkECS)
+    }
+    
+    private func setupPhysicsScene(){
+        scene.sceneUpdateDelegate = self
+    }
+    
+    private func getPhysicsComponents(_ arkECS: ArkECS) -> [(Entity, PhysicsComponent)] {
+        return arkECS.getEntities(with: [PhysicsComponent.self]).compactMap { entity in
+            guard let physicsComponent = arkECS.getComponent(ofType: PhysicsComponent.self, for: entity) else {
+                return nil
+            }
+            return (entity, physicsComponent)
+        }
+    }
+    
+    func syncFromPhysicsEngine(arkECS: ArkECS) {
+        let syncStrategies: [ComponentSyncing] = [PhysicsComponentSync(), PositionComponentSync(), RotationComponentSync()]
+        
+        scene.forEachEntity(perform: { entityId, physicsBody in
+            syncStrategies.forEach { strategy in
+                strategy.sync(entityId: entityId, with: physicsBody, using: arkECS)
+            }
+        })
+    }
+    
+    func syncToPhysicsEngine(_ physicsComponents: [(Entity, PhysicsComponent)], arkECS: ArkECS) {
+        for (entity, physics) in physicsComponents {
+            handlePhysicsComponentRemovalIfNeeded(for: entity, using: physics, arkECS: arkECS)
+            
+            if physics.toBeRemoved {
+                continue
+            }
+            
+            let position = arkECS.getComponent(ofType: PositionComponent.self, for: entity)
+            let rotation = arkECS.getComponent(ofType: RotationComponent.self, for: entity)
+            
+            guard let positionComponent = position, let rotationComponent = rotation else {
+                continue
+            }
+            
```

Comment:
This is probably more readable and idiomatic
```
        guard !physics.toBeRemoved else { continue }

        guard let positionComponent = arkECS.getComponent(ofType: PositionComponent.self, for: entity),
              let rotationComponent = arkECS.getComponent(ofType: RotationComponent.self, for: entity) else {
            continue
        }
```

Reply:
Changed to this!